lib/src/models/parsed_certificate.dart

```diff
@@ -11,66 +12,78 @@ sealed class ParsedCertificate with _$ParsedCertificate {
   const ParsedCertificate._();
 
   const factory ParsedCertificate.stakeRegistration({
+    required CertificateType type,
```

Comment:
Aici nu e bine. Pentru ca tipul certificatului este de fapt constant pentru fiecare factory type.

Cu implementarea sugerata de voi pot crea:
```dart
final stakeCert = ParsedCertificate.stakeRegistration(
  type: CertificateType.stakeDeregistration,
  stakeCredential: ...,
);
```

care ar fi clar incorect (si un caz care nu trebuie sa existe in realitate)

Ce trebuie sa faceti e sa:
1) Stergeti acest `required CertificateType type` pe care l-ati adaugat la toate factory-urile
2) Adaugati la final un field:

```dart
late final type = switch(this) {
  StakeRegistration() => CertificateType.stakeRegistration,
  StakeDeregistration() => CertificateType.stakeDeregistration,
  ...
}
```

Reply:
Am observat acum ca aceeasi greseala exista pentru `StakingDataSource`. Si acolo trebuie corectat. Cautati va rog daca mai exista in alta parte aceeasi greseala.

Reply:
Cred ca astea sunt toate cazurile in care faceti asta: 
`ParsedCertificate`, `ParsedCVoteDelegation`, `ParsedCredential`, `ParsedDRep`
, `ParsedOutputDestination`, `ParsedPoolKey`, `ParsedPoolOwner`, `ParsedPoolRelay`, `ParsedPoolRewardAccount`, `ParsedRequiredSigner`, `ParsedTxAuxiliaryData`, `SpendingDataSource`, `StakingDataSource`,

Reply:
done